pages/_app/index.jsx

```diff
@@ -21,12 +26,40 @@ class App extends NextApp {
     }
   }
 
-  store = null
+  redirectToLogin = () => {
+    window.location.replace('/login.html')
+  }
 
-  componentDidMount() {
+  reflectStoreToRoute() {
+    const { store } = this
     const { router } = this.props
-    const { dataProvider, activity } = new Route(router.asPath)
-    this.store = initStore({ dataProvider, activity })
+
+    if (store == null) return
+
+    const route = new Route({
+      dataProvider: store.dataProvider?.id,
+      activity: store.activity?.path,
+    })
+
+    // TODO: This probably should be under condition
+    router.push(route.href, route.as)
+  }
+
+  async componentDidMount() {
+    const store = initStore()
+
+    // Assumes store object is always the same
+    this.store = store
+    this.disposeRouter = autorun(this.reflectStoreToRoute.bind(this))
+
+    try {
+      const { dataProvider, activity } = new Route(window.location.pathname)
+      await store.init(dataProvider, activity)
+      this.setState({ isAuthorized: true })
+    } catch (unauthorizedError) {
```

Comment:
This error never will never happen I think.

Reply:
Yes but it should. If user is not authorised, cookie changes or something, we get 500. It's more API problem, we can expect 401.

Reply:
Sorry I didn't notice this return statement.

https://github.com/oacore/dashboard/blob/login-debugging/api/index.js#L34

Reply:
Honestly, I forgot to remove that part and throw error instead.

Reply:
I've just did it.